src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticQuerySource.cs

```diff
@@ -78,10 +81,17 @@ public async Task<IQueryResults> ExecuteQueryAsync(Query query, IDictionary<stri
         {
             var results = new List<JsonObject>();
 
-            foreach (var document in docs.TopDocs)
+            foreach (var document in docs.Hits)
```

Comment:
Why `Hits` here not `TopDocs`?

Reply:
Not good.

Reply:
Hits contain all the data I need (including _score and highlights), topDocs is the information that contains in Hits.Source.

Reply:
you should be able to use return `_score` and `highlights` in the TopDocs

Reply:
So, what should I do? Should I correct topDocs to fill it from elasticSearch.Hits? Or I can leave it as that and just add extra info to graphQL?

Reply:
> Not good. This is the old way of returning results. You should use TopDocs now. This will alter all the search results.

For now - it will not affect any data, espicially if you choose to return contentItems. It just adds _score and highlight to elasticsearch query, when you selected return type in your elasticsearch query as object

Reply:
`{
    "type": "object",
    "parameters": {
        "first": "Integer",
        "query": "String"
    },
    "properties": {
        "Content.ContentItem.FullText": {
            "type": "string",
            "description": "FullText"
        },
        "ContentItem.ContentType": {
            "type": "string",
            "description": "Type"
        },
        "Highlight": {
            "type": "string",
            "description": "highlight"
        },
        "_score": {
            "type": "string",
            "description": "score"
        }
    }
}`

Reply:
@denispetrische I think you should modify the `ElasticQueryService` class, before the line "hits.Add(row);" you could add 

```
row["_source"] = hit.Source;
row["highlight"] = hit.Highlight;
```

Then in the source you'll have that info available to be returned. 

But, if you add highlight and score, you'll need to probably also use them when returning ContentItems. I am wondering if we can return a new object instead of returning content item directly something like

```
public class ElasticsearchQueryResult
{
    public string Key { get; set; }

    public double? Score { get; set; }

    public string Highlight { get; set; }

    public IEnumerable<ContentItem> ContentItems { get; set; }
}
```

But I am not sure about the extend of adding `ElasticsearchQueryResult` instead of returning ContentItems because it likely to require changes to `IQueryResults`

Reply:
---> I think you should modify the ElasticQueryService class, before the line "hits.Add(row);" you could add

![image](https://github.com/user-attachments/assets/251c0659-c4ba-460f-a445-b9a2014bf643)
But this will work only if we specify Fields

![image](https://github.com/user-attachments/assets/27a61cee-81cb-440b-8d9c-2718a664c489)

Reply:
---> But, if you add highlight and score, you'll need to probably also use them when returning ContentItems.
---> But I am not sure about the extend of adding ElasticsearchQueryResult instead of returning ContentItems because it likely to require changes to IQueryResults

Yeah, It's easier to add _score and highlight only for object return type). Will it be okay if I leave contentItems as that, without score and highlight?

I think there is no problem to get all the info I need with object return type.